components/Menu.jsx

```diff
@@ -47,69 +47,39 @@ const adminItems = [
   },
 ];
 
-export default function Menu({ toggleMenu }) {
-  const categories = useStore(state => state.categories);
+function MenuList(item) {
   const { asPath } = useRouter();
+  return (
+    <ListItem key={item.id} disablePadding selected={asPath === item.href}>
+      <Link href={item.href}>
+        <ListItemButton>
+          <ListItemIcon>{item.icon}</ListItemIcon>
+          <ListItemText>{item.text}</ListItemText>
+        </ListItemButton>
+      </Link>
+    </ListItem>
+  );
+}
+
+export default function Menu() {
+  const categories = useStore(state =>
+    state.categories?.map(category => {
+      return {
+        id: category.slug,
+        href: `/categories/${category.slug}`,
+        text: category.name,
+        icon: null,
+      };
+    })
+  );
 
   return (
-    <Box
-      role="presentation"
-      onClick={toggleMenu}
-      onKeyDown={toggleMenu}
-      sx={{ minWidth: 250 }}
-    >
-      <List>
-        {mainItems.map(item => (
-          <ListItem
-            key={item.id}
-            disablePadding
-            selected={asPath === item.href}
-          >
-            <Link href={item.href}>
-              <ListItemButton>
-                {item.icon && <ListItemIcon>{item.icon}</ListItemIcon>}
-                <ListItemText>{item.text}</ListItemText>
-              </ListItemButton>
-            </Link>
-          </ListItem>
-        ))}
-      </List>
+    <Box role="presentation" sx={{ minWidth: 250 }}>
+      <List>{mainItems.map(MenuList)}</List>
```

Comment:
The name MenuList suggests that it returns a complete list, but it returns a ListItem. Renaming it accordingly would increase the readability of your code.

Reply:
Also it is uncommon to call jsx components as normal functions, even though it is possible. I would suggest that you stick to the more common way of writing the code so others it is more accessible to other programmers.